src/Account/QueryHandler/GetContextHandler.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License version 3.0
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
+ */
+
+namespace PrestaShop\Module\PsAccounts\Account\QueryHandler;
+
+use PrestaShop\Module\PsAccounts\Account\Query\GetContextQuery;
+use PrestaShop\Module\PsAccounts\Provider\ShopProvider;
+use PrestaShop\Module\PsAccounts\Service\Accounts\AccountsException;
+
+class GetContextHandler
+{
+    /**
+     * @var ShopProvider
+     */
+    private $shopProvider;
+
+    /**
+     * @param ShopProvider $shopProvider
+     */
+    public function __construct(
+        ShopProvider $shopProvider
+    ) {
+        $this->shopProvider = $shopProvider;
+    }
+
+    /**
+     * @param GetContextQuery $query
+     *
+     * @return array
+     *
+     * @throws AccountsException
+     */
+    public function handle(GetContextQuery $query)
+    {
+        return [
+            // TODO: add the real data
+            'ps_accounts' => [
+                'last_succeeded_upgrade_version' => null,
+                'module_version_from_files' => null,
+            ],
+            'groups' => $this->shopProvider->getShops(), // TODO: pass param to force refresh status ?
```

Comment:
We can pass group and shop ID too

Reply:
Agree query might hold thoses parameters